engines/registry/lib/registry/engine.rb

```diff
@@ -1,6 +1,32 @@
+require 'fileutils'
+
 module Registry
   class Engine < ::Rails::Engine
     isolate_namespace Registry
     config.generators.api_only = true
+
+    config.after_initialize do
+      Api::Connect::V3::Systems::ActivationsController.class_eval do
+        before_action :handle_cache, only: %w[index]
+
+        def handle_cache
+          # get request header
+          if request.headers['X-Instance-Data'] && ZypperAuth.verify_instance(request, logger, @system)
```

Comment:
Is it ok to fail silently here when `ZypperAuth.verify_instance` fails? Probably you want to raise an exception, so that the client can show an error in this case?